jujugui/static/gui/src/app/views/topology/service.js

```diff
@@ -151,10 +152,32 @@ YUI.add('juju-topology-service', function(Y) {
       } else if (d.highlighted) {
         rerenderRelations = true;
       } else {
-        curr_node.attr({
-          'stroke': '#888888',
-          'stroke-width': 1
+        var units = d.units._items;
+        units.forEach(function(unit) {
+          if (unit.agent_state === 'installing'
+              || unit.agent_state === 'pending') {
+            is_pending = true;
+          } else if (unit.agent_state === 'error') {
+            is_erroring = true;
+          }
         });
+
+        if (is_erroring) {
```

Comment:
I think the logic is pretty good for this.  I wonder if it would be easier to add a class to the service blocks rather than setting the stroke color/width as attributes?  That would make it a little easier to test, because then you can just test for the presence of the classes (normal + status-normal, etc).